lib/practice_exercises.rb

```diff
@@ -1,13 +1,47 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: 0(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  j = 1
+  until list[j] == nil
+    if list[i] == list[j]
+      list.delete_at(list[j])
+    else
+      i+=1
+      j+=1
+    end
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^4)
+# Space Complexity: 0(n)
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+  i = 1
+  j = 0
+  k = 0
+  output = ""
+  characters = []
+  length = strings.length
+  strings.each do |word|
+    characters << word.chars
+  end
+  
+  characters.each do |item|
```

Comment:
This isn't working, take a look at the loops a bit.  You can fairly comfortably do this by having one loop go through all the words, and the 2nd looping through the letters in the 1st word.  When you find a letter than not all the words have in the right position, you've found your prefix.